llvm/lib/Support/CommandLine.cpp

```diff
@@ -2734,6 +2734,39 @@ void cl::PrintHelpMessage(bool Hidden, bool Categorized) {
     CommonOptions->CategorizedHiddenPrinter.printHelp();
 }
 
+ArrayRef<StringRef> cl::CompilerBuildConfig = {
```

Comment:
I footgunned here, this doesn't do lifetime extension so this array is a temporary despite being a global. Making this a function to additionally avoid having a static constructor.